aws-pcluster stacks: Use only static spack.yaml

stephenmsachs opened the pull request:
Instead of creating compilers.yaml as needed use a static compilers.yaml for the
aws-pcluster build caches. These entries need to adapted once the container
image changes, but can be used as static objects within spack to enable testing
changes in compilers.yaml.

Also compiler:environment:prepend_path:PATH entry has been removed and set into
$PATH directly.

-------

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

Commit:
Use static compilers: entry in spack.yaml

Instead of creating compilers.yaml as needed use a static compilers.yaml for the
aws-pcluster build caches. These entries need to adapted once the container
image changes, but can be used as static objects within spack to enable testing
changes in compilers.yaml.

Also compiler:environment:prepend_path:PATH entry has been removed and set into
$PATH directly.

Commit:
aws pcluster stacks: Move packages to spack.yaml

Unify stack into one single configuration file and remove the dynamic
substitution of libfabric and SLURM version numbers. These need to be adapted
once the build stack container changes.

stephenmsachs commented:
Also move the `packages.yaml` directly into `spack.yaml` such that there is one single static configuration file for the stack.

stephenmsachs commented:
CI works, stil lneed to clean this up.

haampie commented on share/spack/gitlab/cloud_pipelines/scripts/pcluster/setup-pcluster.sh:
```diff
@@ -108,5 +108,5 @@ install_compilers() {
 set_pcluster_defaults
-install_compilers
-patch_compilers_yaml
+# install_compilers
```
can this file be deleted?

stephenmsachs replied:
Yes, that's part of the cleanup I still need to do.

Commit:
Test: Add New palace version to trigger a C++ build

Commit:
Completely remove file setup-pcluster.sh

Also remodel pcluster entry in .gitlab-ci.yaml since most of "before-script" is
no longer needed.

Commit:
Add setup-env.sh back since this not covered elsewhere

Commit:
Make correct binutils available in build jobs

Commit:
Test: Workaround https://github.com/awslabs/palace/issues/257

Commit:
Revert "Test: Workaround https://github.com/awslabs/palace/issues/257"

This reverts commit 5f77531372c43c45f5fc8492d33490528c8687d7.

Commit:
Test: Workaround xsmm issue

stephenmsachs commented:
Pipeline https://gitlab.spack.io/spack/spack/-/pipelines/910916 compiled new version of `palace`, i.e. C++ code with multipel dependencies using `%oneapi`. Thus showing that the correct `ld` and `libstdc++` are found. 

I'll remove the new `palace` entry and then this is ready for merge.

Commit:
Revert "Test: Workaround xsmm issue"

This reverts commit e5933cf1f28e48f35bbe469a0da774a902a64836.

Commit:
Revert "Test: Add New palace version to trigger a C++ build"

This reverts commit 51823d577b049ba4fefc5adc9b847629fb4c3c8b.

Commit:
Concretizer does not need path to `ld`

Commit:
Revert "Concretizer does not need path to `ld`"

This reverts commit 888329da4b68925806e3ecb13cf4fb5d0829b2d4.
